projects/make-recall-decisions-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/RecommendationService.kt

```diff
@@ -36,20 +36,45 @@ class RecommendationService(
         username: String,
         occurredAt: ZonedDateTime
     ) {
-        val contact = personRepository.getPerson(crn).addContact(
-            details,
-            date = occurredAt,
-            staff = staffRepository.getStaff(username),
-            type = contactTypeRepository.getByCode(ContactType.MANAGEMENT_OVERSIGHT_RECALL),
-            outcome = contactOutcomeRepository.getByCode(decision.code)
-        )
+        val contact = addContact(crn, decision, details, username, occurredAt, ContactType.MANAGEMENT_OVERSIGHT_RECALL)
         contactRepository.save(contact)
         telemetryService.trackEvent(
             "ManagementOversightCreated",
             mapOf("CRN" to crn, "username" to username, "decision" to decision.name)
         )
     }
 
+    fun consideration(
+        crn: String,
+        details: RecommendationDetails,
+        username: String,
+        occurredAt: ZonedDateTime
+    ) {
+        val contact = addContact(crn, null, details, username, occurredAt, ContactType.CONSIDERATION)
+        contactRepository.save(contact)
+        telemetryService.trackEvent(
+            "ConsiderationCreated",
+            mapOf("CRN" to crn, "username" to username)
+        )
+    }
+
+    private fun addContact(
```

Comment:
Minor thing - but could combine this with the Person.addContact function at the bottom of the class?

Reply:
Think they both do different things. The original one doesnt actual add anyting (it gets a contact using person) so ive changed that to getPerson to make it clearer